### ISSUE TITLE
JAX-RS OPTIONS path with regex hides other methods

### ISSUE THREAD

**Issue by peraxel** [payara/Payara#2378]
# Description #
----------

When adding a method to handle OPTIONS-requests with regex in path annotation on an JAX-RS endpoint like below, other methods are hidden and any request to them results in a "405 Method not allowed" response.

```
    @OPTIONS
    @Path("{id: \\w+}")
    public String options(@PathParam("id") String id) {
        return "options " + id;
    }
```

I included three endpoints (test1, test2, test3) in the repo declared below.
Test fails in endpoint test1, but succeeds on test2 and test3.
test1 uses regex in OPTIONS path, but no regex in GET path
test2 and test3 uses regex both OPTIONS and GET paths.

## Expected Outcome

Request: curl -v http://localhost:8080/payara-path-bug/test1/bob
Expected: 200 OK "hello world bob"

## Current Outcome

Request: curl -v http://localhost:8080/payara-path-bug/test1/bob
Current outcome: 405 Method Not Allowed

## Steps to reproduce (Only for bug reports) 

1. Clone Github repository https://github.com/peraxel/payara-path-bug
2. mvn package
3. Deploy to Payara
4. curl -v http://localhost:8080/payara-path-bug/test1/bob

### Samples

https://github.com/peraxel/payara-path-bug

## Context (Optional)

Trying to set up a generic OPTIONS method to catch OPTIONS requests to the endpoint path and all sub-paths.

## Environment ##

- **Payara Version**: 4.1.2.174
- **Edition**: Full
- **JDK Version**: Oracle 1.8.0
- **Operating System**: Mac

**Comment by mulderbaba** [comment]
@peraxel thanks for the detailed input. Your example validates the problem but this is a bug that needs to be fixed in Jersey impl [[#3411](https://github.com/jersey/jersey/issues/3411)] as I see it. If you request your wadl through something like ``http://localhost:8080/payara-path-bug/application.wadl`` you will get a response as:
```
<resource path="test1">
   <resource path="{id: \w+}">
      <param xmlns:xs="http://www.w3.org/2001/XMLSchema" name="id" style="template" type="xs:string" />
      <method id="options" name="OPTIONS">
         <response>
            <representation mediaType="*/*" />
         </response>
      </method>
   </resource>
   <resource path="{id}">
      <param xmlns:xs="http://www.w3.org/2001/XMLSchema" name="id" style="template" type="xs:string" />
      <method id="get2" name="GET">
         <response>
            <representation mediaType="*/*" />
         </response>
      </method>
   </resource>
</resource>
```
So ``test1`` resources first maps to ``"{id: \w+}"``. What you can do here as a workaround is: change the annotations of ``get2()`` method as:
```
    @GET
    @Path("{id: \\w+}")
    public String get2(@PathParam("id") String id) {
        return "hello world " + id;
    }
```
so that you will provide both ``GET`` and ``OPTIONS`` for ``"{id: \w+}"``.

**Comment by smillidge** [comment]
If it's a Jersey bug we can look to fix when that moved to Eclipse as part of the EE4J project

**Comment by OndroMih** [comment]
Hi @peraxel, I tried your reproducer on the latest Payara Server 5.201 and I see the expected behavior `200 OK "hello world bob"`. Can I close this issue?

**Comment by OndroMih** [comment]
Hi @peraxel, do you still have this issue in 5.201? I'd like to close this if you don't have issues anymore.

**Comment by peraxel** [comment]
@OndroMih Confirmed fixed in 5.201
